src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,395 @@
+            }
+            catch (ArgumentException argException)
+            {
+                WriteError(new ErrorRecord(argException, "FormatHexArgumentError", ErrorCategory.WriteError, path));
+            }
+            catch (NotSupportedException notSupportedException)
+            {
+                WriteError(new ErrorRecord(notSupportedException, "FormatHexPathRefersToANonFileDevice", ErrorCategory.InvalidArgument, path));
+            }
+            catch (SecurityException securityException)
+            {
+                WriteError(new ErrorRecord(securityException, "FormatHexUnauthorizedAccessError", ErrorCategory.PermissionDenied, path));
+            }
+        }
+
+        #endregion
+
+        #region InputObjects
+
+        /// <summary>
+        /// Returns a new instance of the Encoding Class based on cmdlet Encoding parameter.
+        /// </summary>
+        /// <param name="encoding"></param>
+        /// <returns></returns>
+        private Encoding GetEncoding(string encoding)
+        {
+            if (String.Equals(encoding, "BigEndianUnicode", StringComparison.OrdinalIgnoreCase))
+            {
+                return System.Text.Encoding.BigEndianUnicode;
+            }
+            else if (String.Equals(encoding, "Unicode", StringComparison.OrdinalIgnoreCase))
+            {
+                return System.Text.Encoding.Unicode;
+            }
+            else if (String.Equals(encoding, "UTF7", StringComparison.OrdinalIgnoreCase))
+            {
+                return System.Text.Encoding.UTF7;
+            }
+            else if (String.Equals(encoding, "UTF8", StringComparison.OrdinalIgnoreCase))
+            {
+                return System.Text.Encoding.UTF8;
+            }
+            else if (String.Equals(encoding, "UTF32", StringComparison.OrdinalIgnoreCase))
+            {
+                return System.Text.Encoding.UTF32;
+            }
+            else
+            {
+                return System.Text.Encoding.ASCII;
+            }
+        }
+
+        /// <summary>
+        /// Creates a byte array from the object passed to the cmdlet (based on type) and passes
+        /// that array on to the ConvertToHexidecimal method to output.
+        /// </summary>
+        /// <param name="_inputObject"></param>
+        private void ProcessObjectContent(PSObject _inputObject)
+        {
+            Object obj = _inputObject.BaseObject;
+
+            if (obj is System.IO.FileSystemInfo)
+            {
+                string[] path = { ((FileSystemInfo)obj).FullName };
+                List<string> pathsToProcess = ResolvePaths(path, true);
+                ProcessPath(pathsToProcess);
+            }
+            
+            else if (obj is string)
+            {
+                string inputObject = obj.ToString();
+                byte[] inputBytes = GetEncoding(_encoding).GetBytes(inputObject);
+                ConvertToHexidecimal(inputBytes, null, 0);
```

Comment:
Looks like you could get the inputBytes out of each of these else if {} blocks and do the convert* at the end if else block. that would prevent 10 uses of convert calls peppered across this method.